values-schema.yaml

```diff
@@ -609,6 +703,120 @@ definitions:
   url:
     pattern: '^(https:\/\/)([\w\-])+\.{1}([a-zA-Z]{2,63})([\/\w-]*)*\/?\??([^#\n\r]*)?#?([^\n\r]*)$'
     type: string
+  vaultPath:
+    description: A path to a secret at Vault KV engine
+    pattern: '^\/([A-z0-9-_+]+\/)*([A-z0-9-_]+)$'
+    type: string
+  vaultProperty:
+    description: A property name at vaultPath
+    pattern: '^[a-zA-Z0-9_]*$'
+    type: string
+  vaultDefaultSeal:
+    properties:
+      type:
+        type: string
+        default: default
+        enum:
+          - default
+  vaultAwsStorage:
+    properties:
+      region:
+        type: string
+      bucket:
+        type: string
+    required:
+      - region
+      - bucket
+  vaultAwsSeal:
+    properties:
+      type:
+        type: string
+        default: awskms
+        enum:
+          - awskms
+      region:
+        type: string
+      access_key:
+        type: string
+      secret_key:
+        type: string
+      kms_key_id:
+        type: string
+      endpoint:
+        type: string
+    required:
+      - type
+      - region
+      - access_key
+      - secret_key
+      - kms_key_id
+      - endpoint
+  vaultAzureStorage:
+    properties:
+      accountName:
+        type: string
+      accountKey:
+        type: string
+      container:
+        type: string
+      environment:
+        type: string
+  vaultAzureSeal:
+    properties:
+      type:
+        type: string
+        default: azurekeyvault
+        enum:
+          - azurekeyvault
+      vault_name:
+        type: string
+      key_name:
+        type: string
+      tenant_id:
+        type: string
+      client_id:
+        type: string
+      client_secret:
+        type: string
+    required:
+      - type
+      - vault_name
+      - key_name
+      - tenant_id
+      - client_id
+      - client_secret
+
+  vaultGoogleStorage:
+    properties:
+      bucket:
+        type: string
+    required:
+      - bucket
+  vaultGoogleSeal:
+    properties:
+      type:
+        type: string
+        default: gcpckms
+        enum:
+          - gcpckms
+      project:
+        type: string
+      region:
+        type: string
+      key_ring:
+        type: string
+      crypto_key:
+        type: string
+      encodedCloudToken:
+        type: string
+    required:
+      - type
+      - project
+      - region
+      - key_ring
+      - crypto_key
+      - encodedCloudToken
```

Comment:
I don't think these need to be re-usable props here as they are only used in one location.

Reply:
It is much easier to read complex schemas if you split them into smaller pieces, therefore I would like to keep it like that.

Reply:
done